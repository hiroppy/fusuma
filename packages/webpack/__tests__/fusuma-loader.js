'use strict';

const mdx = require('@mdx-js/mdx');
const plugin = require('../src/fusumaMdxPlugin');
const loader = require('../src/fusuma-loader');

async function transformToJS(src) {
  return new Promise((resolve, reject) => {
    loader.call(
      {
        async: () => (err, res) => {
          if (err) {
            return reject(err);
          }
          resolve(res);
        }
      },
      src
    );
  });
}

describe('fusuma-loader', () => {
  test('should return normal md', async () => {
    const src = `
# 1
---
# 2
`;
    expect(await transformToJS(src)).toMatchSnapshot();
  });

  test('should return normal mdx', async () => {
    const src = `
import React from 'react';

const Sample = () => <div>sample</div>;

<Sample />
`;
    expect(await transformToJS(src)).toMatchSnapshot();
  });

  test('should add fusuma options', async () => {
    const src = `
<!-- note
This is Note!
-->
<!-- classes: class! -->
<!-- sectionTitle: sectionTitle! -->
# Hello
<!-- content -->
<!-- screen -->
`;

    expect(await transformToJS(src)).toMatchSnapshot();
  });

  test('should convert emoji', async () => {
    const src = `
# :smile:
---
# :smile:
`;

    expect(await transformToJS(src)).toMatchSnapshot();
  });

  test('should convert mermaid', async () => {
    const src = `
# FlowChart
---

\`\`\`flow
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
\`\`\`
`;

    expect(await transformToJS(src)).toMatchSnapshot();
  });
});
