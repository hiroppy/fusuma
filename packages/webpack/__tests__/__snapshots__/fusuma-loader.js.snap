// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fusuma-loader should add fusuma options 1`] = `
"/* @jsx mdx */


        import React from 'react';
        import { mdx } from '@mdx-js/react';
        export const slides = [
          (props) => (
            <>
              
    {
      /* note
      This is Note!
      */
    }
    {
      /* classes: class! */
    }
    {
      /* sectionTitle: sectionTitle! */
    }
    <h1>{\`Hello\`}</h1>
    {
      /* content */
    }
    {
      /* screen */
    }
    <div className=\\"fusuma-screen\\"><div>This view can capture the screen.<br />Click to get started.;)<br /><br />Note: This feature runs only in Presenter Mode.</div><video id=\\"fusuma-screen-1\\" /></div>
    
            </>
          )];
        export const fusumaProps = [{note: 'This is Note!',classes: 'class!',sectionTitle: 'sectionTitle!',screen: 'true'}];
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  slides
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    {
      /* note
      This is Note!
      */
    }
    {
      /* classes: class! */
    }
    {
      /* sectionTitle: sectionTitle! */
    }
    <h1>{\`Hello\`}</h1>
    {
      /* content */
    }
    {
      /* screen */
    }

    </MDXLayout>;
}

MDXContent.isMDXComponent = true;"
`;

exports[`fusuma-loader should convert emoji 1`] = `
"/* @jsx mdx */


        import React from 'react';
        import { mdx } from '@mdx-js/react';
        export const slides = [
          (props) => (
            <>
              
    <h1>{\`ðŸ˜„\`}</h1>
    
            </>
          ),

          (props) => (
            <>
              
    <h1>{\`ðŸ˜„\`}</h1>
    
            </>
          )];
        export const fusumaProps = [{},
{}];
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  slides
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    <h1>{\`ðŸ˜„\`}</h1>
    <hr></hr>
    <h1>{\`ðŸ˜„\`}</h1>

    </MDXLayout>;
}

MDXContent.isMDXComponent = true;"
`;

exports[`fusuma-loader should convert mermaid 1`] = `
"/* @jsx mdx */


        import React from 'react';
        import { mdx } from '@mdx-js/react';
        export const slides = [
          (props) => (
            <>
              
    <h1>{\`FlowChart\`}</h1>
    
            </>
          ),

          (props) => (
            <>
              
    <pre><code parentName=\\"pre\\" {...{
        \\"className\\": \\"language-flow\\"
      }}>{\`graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
\`}</code></pre>
    
            </>
          )];
        export const fusumaProps = [{},
{}];
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  slides
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    <h1>{\`FlowChart\`}</h1>
    <hr></hr>
    <pre><code parentName=\\"pre\\" {...{
        \\"className\\": \\"language-flow\\"
      }}>{\`graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
\`}</code></pre>

    </MDXLayout>;
}

MDXContent.isMDXComponent = true;"
`;

exports[`fusuma-loader should return normal md 1`] = `
"/* @jsx mdx */


        import React from 'react';
        import { mdx } from '@mdx-js/react';
        export const slides = [
          (props) => (
            <>
              
    <h1>{\`1\`}</h1>
    
            </>
          ),

          (props) => (
            <>
              
    <h1>{\`2\`}</h1>
    
            </>
          )];
        export const fusumaProps = [{},
{}];
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};

const layoutProps = {
  slides
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">
    <h1>{\`1\`}</h1>
    <hr></hr>
    <h1>{\`2\`}</h1>

    </MDXLayout>;
}

MDXContent.isMDXComponent = true;"
`;

exports[`fusuma-loader should return normal mdx 1`] = `
"/* @jsx mdx */
import React from 'react';

        import React from 'react';
        import { mdx } from '@mdx-js/react';
        export const slides = [
          (props) => (
            <>
              

    <p>{\`const Sample = () => \`}<div>{\`sample\`}</div>{\`;\`}</p>
    <Sample mdxType=\\"Sample\\" />
    
            </>
          )];
        export const fusumaProps = [{}];
const makeShortcode = name => function MDXDefaultShortcode(props) {
  console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
  return <div {...props}/>
};
const Sample = makeShortcode(\\"Sample\\");
const layoutProps = {
  slides
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">

    <p>{\`const Sample = () => \`}<div>{\`sample\`}</div>{\`;\`}</p>
    <Sample mdxType=\\"Sample\\" />

    </MDXLayout>;
}

MDXContent.isMDXComponent = true;"
`;
